### PR TITLE
Integrating OpenSSL 3 Changes and Fixing Pipelines for Ubuntu 22.04 

### PR DESCRIPTION
**Pull request by nihemstr**
- Changed install-deps to effectively manage the openssl version
- Added compile time support for OpenSSL 3
- Added the install-openssl.sh script for customer and engineer installation easy install step

### CODE REVIEW

File: tools/openssl_install_script/install-openssl.sh [hunk 1]
```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# install-deps.sh makes it more convenient to install
```
Comment (jw-msft) [inline tools/openssl_install_script/install-openssl.sh:6]: nit: install-openssl.sh

**Review by jw-msft — APPROVED** [review]
approved with some nits.

File: tools/openssl_install_script/install-openssl.sh [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    elif [[ $openssl_major_version -eq "3" ]]; then
+        # Need to handle sym linking 3.x
+        openssl_so_suffix="3"
+    else
+        error "OpenSSL version $openssl_version is not supported"
+        error "Only installations of 1.1 and 3.x are supported"
+        $ret 1
+    fi
+
+    ## if the openssl version is 3.1 it's
```
[the remark this answers is not in the document]
Comment (nihemstr) [reply tools/openssl_install_script/install-openssl.sh:92]: Running too fast and not checking what I'm writing caused some of these errors. Re reading and editing so my frenetic comments actually make sense.

**Review by jw-msft — DISMISSED** [review]
stoked that we were able to remove openssl 1.1.1 completely. Nicely done.

**Review by jw-msft — APPROVED** [review]
This time for sure.

**Pull request merged**